### DISCUSSION TITLE
Contribution guidelines to onboard new developers

### DISCUSSION THREAD

**Discussion by dhruvkb** [ElMassimo/iles#137] [General]
I really love using iles after switching over from Astro. One thing that strikes me is that the documentation here is fairly minimal. While certainly enough to get started and helpful in many places, it's very sparse. I would like to help but there are no contribution guidelines for contributing code or docs.

If the project is looking to add more contributors, a small guide to help set up a dev or docs environment would be a great place to start.

**Comment by ElMassimo** [comment]
Hi Druv! 

I like the suggestion, we could add a `CONTRIBUTING.md` file based on https://github.com/antfu/contribute.

- This project is using `pnpm` as the package manager
- Running `pnpm run build:all` will build all packages and modules in the monorepo
- Running `pnpm run docs` will start the docs website in development mode
- Running `pnpm run blog` will start the Vue blog example in development mode

---

I never expected this project to become as popular, and I'm really happy to see that other developers find it useful, and are shipping websites such as [Kia Colombia](https://www.kia.com.co/) 😃 

The current guide is really lacking a more hands-on recipes section, which discusses **specific use cases** in depth, such as:

- [Creating a blog](https://github.com/ElMassimo/iles/tree/main/playground/the-vue-point)
- [Using data from a CMS](https://github.com/ElMassimo/iles/discussions/124)
- [Reading data from Git](https://github.com/ElMassimo/iles/discussions/132)
- [Adding third-party scripts](https://github.com/ElMassimo/iles/discussions/126)
- [Rendering markdown from dynamic sources](https://github.com/ElMassimo/iles/discussions/103)

This content is scattered in the [GitHub Discussions](https://github.com/ElMassimo/iles/discussions), but it would be awesome to create an îles blog with full-blown articles covering these use cases and more, which would make them easy to discover, and a great way to engage the community which can subscribe to the blog and see interesting things that can be done with îles.

I'd also love to have "guest blog posts" from the community in this blog, sharing their experience, tips, and tricks with îles.
